int32 source
int32 target
---
int32[] vertex_list
